python/private/py_console_script_binary.bzl

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implementation for the macro to generate an console_script py_binary from reading the 'entry_points.txt'.
+"""
+
+load("//python:py_binary.bzl", "py_binary")
+load(":py_console_script_gen.bzl", "py_console_script_gen")
+
+def _dist_info(pkg):
+    """Return the first candidate for the dist_info target label.
+
+    We cannot do `Label(pkg)` here because the string will be evaluated within
+    the context of the rules_python repo_mapping and it will fail because
+    rules_python does not know anything about the hub repos that the user has
+    available.
+
+    NOTE: Works with `incompatible_generate_aliases` and without by assuming the
+    following formats:
+        * @pypi_pylint//:pkg
+        * @pypi//pylint
+        * @pypi//pylint:pkg
+    """
+    return pkg.replace(":pkg", "") + ":dist_info"
+
+def py_console_script_binary(
+        *,
+        name,
+        pkg,
+        entry_points_txt = None,
+        script = None,
+        binary_rule = py_binary,
+        **kwargs):
+    """Generate a py_binary for a console_script entry_point.
+
+    Args:
+        name: str, The name of the resulting target.
+        pkg: The package for which to generate the script.
+        entry_points_txt: The file to be used for parsing the available
+            console_script values. Defaults to searching for `entry_points.txt`
+            in the `dist_info` filegroup in the same package as the `pkg`
+            Label.
```

Comment:
```suggestion
        entry_points_txt: optional target, the entry_points.txt file to parse
            for available console_script values. It may be a single file, or
            group of files, but must contain a file named `entry_point.txt`.
            If not specified, defaults to the `dist_info` target in the same
            package as the `pkg` Label.
```

Reply:
Thanks, better wording than mine.